### ISSUE TITLE
Question about commented out line(host to device communication line)

### ISSUE THREAD

**Issue by Dicklessgreat** [will-hart/switchy#1]
I am grateful for this repository because it describes exactly what I was trying to do.

I have a question regarding the part of setting up key macros from the host to the `Switchy` device.
Unless I am missing something, is this feature not implemented yet?
If so, I am also just stuck in the same spot and have managed to transfer data using `usb-device` crate, `usbd-hid` crate and `usbd-serial` crate, but with no success.
Specifically, I try to get the device and endpoint address from the `rusb` crate and then execute `write_bulk()`, `write_interrupt()`, etc. via the handler, but it returns a "Not Found" error.
I am confused as to why this is happening, although I have verified that the endpoints and devices definitely exist.

Back on topic, but if you don't mind, I'd like to know why [usb_interrupt() is commented out,](https://github.com/will-hart/switchy/blob/ad3715c91aa9a78fa9bc7524e055dee4e1bdb450/src/app.rs#L460) and if you have any insight into this area at this time.

I'm using Windows10 and Raspberry Pi Pico.

**Comment by will-hart** [comment]
Hey, thanks for your interest in switchy :) You are right in that host -> device comms aren't implemented yet. (I got very busy for work and haven't touched this for a while).

I've done it in the past over here https://github.com/will-hart/pedalrs/blob/main/src/main.rs#L54 on the device side and https://gist.github.com/will-hart/b49eaed1dd87c80fa6291bf9d71f195a as an example of how I did it on the PC side. I had a simple egui app that was able to send the bytes to the `pedalrs` USB device.

**Comment by Dicklessgreat** [comment]
Thanks!

I have tried a bit, but host keeps saying an "Input/Output Error" [on this line](https://gist.github.com/will-hart/b49eaed1dd87c80fa6291bf9d71f195a#file-usb_device-rs-L186).
It recognize device, interface, and endpoint. But failed to write... seems weird.
Perhaps it is a problem with Pico or Windows. I will try to find out more.

The host program is exactly the same as the Gist you sent.
The pedalrs is for STM32f chip and my Pico is RP2040, so it may take some time to reproduce your code on my environment.
I was wondering if it has anything to do with the fact that there is no `usb::Peripheral` in the rp2040's hal so I'm using `usb::UsbBus` for Allocator instead.
Please forgive my messy questions that do not have much to do with Switchy. Anyway, you have been very helpful. Thank you very much.

**Comment by Dicklessgreat** [comment]
After some trial and error, I found that apparently there was a problem with Windows.
I attempted to mount the keyboard USB device on the WSL and ran the host program you gave me with sudo, and the data was transmitted correctly.
I wonder what is wrong with Windows, but now I know that some other factor but my program prevented me from doing this, I will move on to investigate that.
This has been very informative. Thank you very much.
